apps/aechannel/src/aesc_channels.erl

```diff
@@ -0,0 +1,222 @@
+%%%=============================================================================
+%%% @copyright (C) 2018, Aeternity Anstalt
+%%% @doc
+%%%    ADT for channel objects
+%%% @end
+%%%=============================================================================
+-module(aesc_channels).
+
+-include_lib("apps/aecore/include/common.hrl").
+
+%% API
+-export([deserialize/1,
+         deposit/2,
+         is_active/1,
+         is_solo_closed/2,
+         is_solo_closing/2,
+         new/1,
+         peers/1,
+         serialize/1,
+         slash/3,
+         close_solo/3,
+         withdraw/2]).
+
+%% Getters
+-export([id/1,
+         id/3,
+         initiator/1,
+         responder/1,
+         total_amount/1,
+         initiator_amount/1,
+         responder_amount/1,
+         channel_reserve/1,
+         round/1,
+         closes_at/1]).
+
+-compile({no_auto_import, [round/1]}).
+
+%%%===================================================================
+%%% Types
+%%%===================================================================
+
+-type id()     :: binary().
+-type amount() :: non_neg_integer().
+-type seq_number() :: non_neg_integer().
+
+-record(channel, {id               :: id(),
+                  initiator        :: pubkey(),
+                  responder        :: pubkey(),
+                  total_amount     :: amount(),
+                  initiator_amount :: amount(),
+                  channel_reserve  :: amount(),
+                  round            :: seq_number(),
+                  lock_period      :: non_neg_integer(),
+                  closes_at        :: height()}).
+
+-opaque channel() :: #channel{}.
+
+-type serialized() :: binary().
+
+-export_type([id/0,
+              amount/0,
+              seq_number/0,
+              channel/0,
+              serialized/0]).
+
+-define(CHANNEL_TYPE, <<"channel">>).
+-define(CHANNEL_VSN, 1).
+
+-define(PUB_SIZE, 32).
+-define(NONCE_SIZE, 256).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec close_solo(channel(), aesc_offchain_tx:tx(), height()) -> channel().
+close_solo(#channel{lock_period = LockPeriod} = Ch, State, Height) ->
+    ClosesAt = Height + LockPeriod,
+    Ch#channel{initiator_amount = aesc_offchain_tx:initiator_amount(State),
+               total_amount     = aesc_offchain_tx:initiator_amount(State) + aesc_offchain_tx:responder_amount(State),
+               round            = aesc_offchain_tx:round(State),
+               closes_at        = ClosesAt}.
+
+-spec deposit(channel(), amount()) -> channel().
+deposit(#channel{total_amount = TotalAmount} = Ch, Amount) ->
+    Ch#channel{total_amount = TotalAmount + Amount}.
+
+-spec deserialize(binary()) -> channel().
+deserialize(Bin) ->
+    {ok, List} = msgpack:unpack(Bin),
```

Comment:
Why are channels serialized with msgpack? It should be serialized in the same way as the other objects.

Reply:
Addressed